nototools/compare_cmap_data.py

```diff
@@ -62,33 +84,54 @@ def compare_cmap_data(base_cmap_data, target_cmap_data, scripts, cps,
     if scripts and script not in scripts:
       continue
     if script in target_map:
-      name = base_map[script].name
-      base_cps = lint_config.parse_int_ranges(base_map[script].ranges)
-      target_cps = lint_config.parse_int_ranges(target_map[script].ranges)
-      added, removed = compare_cmaps(
-          base_cps, target_cps, cps, except_cps, no_additions, no_removals)
-      base_xcps, target_xcps = (set(), set())
-      if int(getattr(base_map[script], 'xcount', -1)) != -1:
-        base_xcps = lint_config.parse_int_ranges(base_map[script].xranges)
-      if int(getattr(target_map[script], 'xcount', -1)) != -1:
-        target_xcps = lint_config.parse_int_ranges(target_map[script].xranges)
-      xadded, xremoved = compare_cmaps(
-          base_xcps, target_xcps, cps, except_cps, no_additions, no_removals)
-      result[script] = added, removed, xadded, xremoved
+      base_data = base_map[script]
+      target_data = target_map[script]
+      result[script] = compare_data(base_data, target_data, opts)
   return result
 
 
-def compare_cmap_data_files(base_file, target_file, scripts, ranges,
-                            except_scripts, except_ranges,
-                            no_additions, no_removals):
+def compare_cmap_data_files(
+    base_file, target_file, scripts, except_scripts, opts=None):
   base_cmap_data = cmap_data.read_cmap_data_file(base_file)
   target_cmap_data = cmap_data.read_cmap_data_file(target_file)
   compare = compare_cmap_data(
-      base_cmap_data, target_cmap_data, scripts, ranges,
-      except_scripts, except_ranges, no_additions, no_removals)
+      base_cmap_data, target_cmap_data, scripts, except_scripts, opts)
   return compare, base_cmap_data, target_cmap_data
 
 
+def compare_interscript(base_map, base_script, target_script, opts=None):
+  missing_scripts = []
+  if base_script not in base_map:
+    missing_scripts.append(base_script)
+  if target_script not in base_map:
+    missing_scripts.append(target_script)
+  if missing_scripts:
+    print 'Cannot compare %s and %s, %s not in cmap data.' % (
+        base_script, target_script, ', '.join(missing_scripts))
+    return
+  base_data = base_map[base_script]
+  target_data = base_map[target_script]
+  return compare_data(base_data, target_data, opts)
+
+
+def compare_interscript_data(base_cmap_data, scripts, opts=None):
+  result = {}
+  base_map = cmap_data.create_map_from_table(base_cmap_data.table)
+  for i in range(len(scripts) - 1):
+    base_script = scripts[i]
+    for j in range(i + 1, len(scripts)):
+      target_script = scripts[j]
+      result[(base_script, target_script)] = compare_interscript(
```

Comment:
I guess the standard injunction against unnecessary parentheses applies.  I'll try to get used to it.